src/components/views/rooms/RoomTile.tsx

```diff
@@ -522,6 +533,11 @@ export default class RoomTile extends React.PureComponent<IProps, IState> {
                         label={_t("Settings")}
                         iconClassName="mx_RoomTile_iconSettings"
                     />
+                    <IconizedContextMenuOption
+                        onClick={this.onCopyRoomClick}
+                        label={_t("Copy")}
```

Comment:
Copy -what-?

Even if the icon is obvious this needs to be clearer for aria users

Reply:
We could do *Copy URL* or *Copy Link* - I assume we don't need to use the word "room" as the other menu items don't

Doesn't help aria, but I guess the icon could be of a link?

Reply:
I vouch for "Copy Link", as URL is more specific (and technically not correct with the `matrix:` URI). A link icon with that would work :+1:

Reply:
This also runs the risk of added confusion in general as users might expect this link to work to let people into the room, e.g the links they would similarly copy in Discord. But this link only works for users already joined or for public rooms. Don't think its possible to come up with a short phrase which explains that intricacy though

Reply:
Could we potentially flash a warning message when a private room is copied? Not really keen on a full-on modal dialog that needs to be dismissed, but does the SDK have some unobtrusive toast thing?

Reply:
We have toasts but they're also pretty bulky
(they cover up a lot of interactive elements in the top left)

![image](https://user-images.githubusercontent.com/2403652/125655097-3b155480-3390-42f9-83d6-e78fcbda7347.png)

There is one thing which has some much lighter things in the bottom left, I think room list action failure, but those are so light that I always fail to notice them

Reply:
> This also runs the risk of added confusion in general as users might expect this link to work to let people into the room, e.g the links they would similarly copy in Discord.

I am not sure we should assume that all (or even most) users assume precisely the same semantics as Discord.  If you copy a link to something on the web, it's obvious that it'll only work if you have access to it.  I don't think we need to spell out that this will not magically autoinvite people into the room.

"Copy link" sounds great to me - i assume this will emit a matrix.to link.